### DISCUSSION TITLE
Q: Different executable sizes with same options?

### DISCUSSION THREAD

**Discussion by posguy99** [ksh93/ksh#495] [General]
I'm sure this is me just not actually understanding how `clang` works.  Feel free to label clueless.

Build ksh...

```
$ bin/package clean && bin/package make
$ ll arch/darwin.i386-64/bin/ksh
-rwxr-xr-x  1 mwilson  staff  1292120 Jul 28 14:26 arch/darwin.i386-64/bin/ksh*
```

Ok, that’s nice and small.

Build ksh this way...

```
$ bin/package clean
$ CCFLAGS=-USHOPT_EDPREDICT bin/package make
$ ll arch/darwin.i386-64/bin/ksh
-rwxr-xr-x  1 mwilson  staff  1879776 Jul 28 15:24 arch/darwin.i386-64/bin/ksh*
```

Huh?

Yes, it's disabled in SHOPT.sh, I never changed the makefile I was building ksh with and the -U has been in it forever.  Who cares if I undefine it or not, SHOPT.sh explicitly sets it.

**Comment by McDutchie** [comment]
The build system currently has an annoying characteristic: if you specify any `CCFLAGS`, they override the default flags instead of adding to them. I may get around to changing that sometime.

In any case, with clang and gcc, the default `CCFLAGS` are `-Os` (optimise for small executable size), so if you add that, you should get an identical executable.

**Reply by posguy99** [reply]
I don't NEED to specify disabling that any more, since it's disabled by default.

**Reply by McDutchie** [reply]
Of course. But you wanted to know why your executable was larger. My answer explained that. Because you passed CCFLAGS (any CCFLAGS), the default -Os flag was overridden.

**Reply by posguy99** [reply]
Yea, I didn't mean to be snarky.  Apologies.

**Reply by McDutchie** [reply]
No worries.
